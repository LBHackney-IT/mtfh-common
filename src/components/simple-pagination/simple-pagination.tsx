import React, { ComponentPropsWithoutRef, forwardRef } from 'react';
import cn from 'classnames';
import './styles.scss';

export const SimplePagination = forwardRef<
  HTMLElement,
  ComponentPropsWithoutRef<'nav'>
>(function SimplePagination({ className, ...props }, ref) {
  return (
    <nav
      ref={ref}
      className={cn('lbh-simple-pagination', className)}
      {...props}
    />
  );
});

export interface SimplePaginationButtonProps
  extends ComponentPropsWithoutRef<'button'> {
  title?: string;
  variant: 'previous' | 'next';
}

export const SimplePaginationButton = forwardRef<
  HTMLButtonElement,
  SimplePaginationButtonProps
>(function SimplePaginationButton(
  { variant, className, title, children, ...props },
  ref
) {
  return (
    <button
      type="button"
      ref={ref}
      className={cn(
        'lbh-simple-pagination__link',
        { 'lbh-simple-pagination__link--next': variant === 'next' },
        className
      )}
      {...props}
    >
      {variant === 'previous' ? (
        <svg width="11" height="19" viewBox="0 0 11 19" fill="none">
          <path d="M10 1L2 9.5L10 18" strokeWidth="2" />
        </svg>
      ) : null}
      {children}
      {title ? (
        <span className="lbh-simple-pagination__title">{title}</span>
      ) : null}
      {variant === 'next' ? (
        <svg width="11" height="19" viewBox="0 0 11 19" fill="none">
          <path d="M1 18L9 9.5L1 1" strokeWidth="2" />
        </svg>
      ) : null}
    </button>
  );
});
