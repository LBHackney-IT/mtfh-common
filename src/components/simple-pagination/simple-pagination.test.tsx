import React from 'react';
import { render } from '@testing-library/react';

import { SimplePagination, SimplePaginationButton } from './simple-pagination';

test('it renders the nav with buttons', () => {
  const { container } = render(
    <SimplePagination>
      <SimplePaginationButton variant="previous">Prev</SimplePaginationButton>
      <SimplePaginationButton variant="next">Next</SimplePaginationButton>
    </SimplePagination>
  );
  expect(container).toMatchSnapshot();
});

test('it renders the nav with buttons and titles', () => {
  const { container } = render(
    <SimplePagination>
      <SimplePaginationButton variant="previous" title="1 of 3">
        Prev
      </SimplePaginationButton>
      <SimplePaginationButton variant="next" title="3 of 3">
        Next
      </SimplePaginationButton>
    </SimplePagination>
  );
  expect(container).toMatchSnapshot();
});
