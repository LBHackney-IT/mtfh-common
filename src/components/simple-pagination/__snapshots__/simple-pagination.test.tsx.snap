// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the nav with buttons 1`] = `
<div>
  <nav
    class="lbh-simple-pagination"
  >
    <button
      class="lbh-simple-pagination__link"
      type="button"
    >
      <svg
        fill="none"
        height="19"
        viewBox="0 0 11 19"
        width="11"
      >
        <path
          d="M10 1L2 9.5L10 18"
          stroke-width="2"
        />
      </svg>
      Prev
    </button>
    <button
      class="lbh-simple-pagination__link lbh-simple-pagination__link--next"
      type="button"
    >
      Next
      <svg
        fill="none"
        height="19"
        viewBox="0 0 11 19"
        width="11"
      >
        <path
          d="M1 18L9 9.5L1 1"
          stroke-width="2"
        />
      </svg>
    </button>
  </nav>
</div>
`;

exports[`it renders the nav with buttons and titles 1`] = `
<div>
  <nav
    class="lbh-simple-pagination"
  >
    <button
      class="lbh-simple-pagination__link"
      type="button"
    >
      <svg
        fill="none"
        height="19"
        viewBox="0 0 11 19"
        width="11"
      >
        <path
          d="M10 1L2 9.5L10 18"
          stroke-width="2"
        />
      </svg>
      Prev
      <span
        class="lbh-simple-pagination__title"
      >
        1 of 3
      </span>
    </button>
    <button
      class="lbh-simple-pagination__link lbh-simple-pagination__link--next"
      type="button"
    >
      Next
      <span
        class="lbh-simple-pagination__title"
      >
        3 of 3
      </span>
      <svg
        fill="none"
        height="19"
        viewBox="0 0 11 19"
        width="11"
      >
        <path
          d="M1 18L9 9.5L1 1"
          stroke-width="2"
        />
      </svg>
    </button>
  </nav>
</div>
`;
